src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1620,6 +1910,36 @@ static bool computeNewHueValue(ColorHueTransitionState * p)
     return false;
 }
 
+static uint16_t addEnhancedHue(uint16_t hue1, uint16_t hue2)
+{
+    uint32_t hue32;
+
+    hue32 = static_cast<uint32_t>(hue1);
+    hue32 = static_cast<uint32_t>(hue32 + static_cast<uint16_t>(hue2));
+
+    if (hue32 > MAX_ENHANCED_HUE_VALUE)
+    {
+        hue32 = static_cast<uint32_t>(hue32 - MAX_ENHANCED_HUE_VALUE);
+    }
+
+    return static_cast<uint16_t>(hue32);
+}
+
+static uint16_t subtractEnhancedHue(uint16_t hue1, uint16_t hue2)
```

Comment:
Again, it's not clear to me why this can't just do subtraction on `uint16_t` values and cast to `uint16_t`.